### ISSUE TITLE
Confusing docs about submitting forms in functional tests

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony-docs#9585]
In this page (https://symfony.com/doc/current/testing.html#forms) you can find this example:

```php
$form = $buttonCrawlerNode->form(array(
    'name'              => 'Fabien',
    'my_form[subject]'  => 'Symfony rocks!',
));
```

Is the use of `name` instead of `my_form[name]` ... or the use of `my_form[subject]` instead of `subject` done on purpose? It's a bit confusing to me. Thanks!

**Comment by xabbuh** [comment]
The array keys must match the value of the `name` attribute of the corresponding form. So if we assume the rendering is based on a Symfony form type (which we IMO can), we will usually have names that follow the `my_form[subject] pattern. I think we should just update the other keys accordingly for consistency.

**Comment by javiereguiluz** [comment]
Fixed by #9854.
